porch/controllers/remoterootsync/pkg/controllers/remoterootsyncset/remoterootsync_controller.go

```diff
@@ -136,27 +162,94 @@ func (r *RemoteRootSyncSetReconciler) applyToClusterRef(ctx context.Context, sub
 }
 
 // BuildObjectsToApply config root sync
-func BuildObjectsToApply(subject *api.RemoteRootSyncSet) ([]*unstructured.Unstructured, error) {
+func (r *RemoteRootSyncSetReconciler) BuildObjectsToApply(ctx context.Context, subject *api.RemoteRootSyncSet) ([]*unstructured.Unstructured, error) {
+	// TODO: stop requiring this to be set & source from a deployment instead
+	gcpProjectID := os.Getenv("GCP_PROJECT_ID")
+	imageName := oci.ImageTagName{
+		Image: "us-west1-docker.pkg.dev/" + gcpProjectID + "/deployment/myfirstnginx",
+		Tag:   "v1",
+	}
+
+	digest, err := r.ociStorage.LookupImageTag(ctx, imageName)
+	if err != nil {
+		return nil, err
+	}
+
+	resources, err := r.ociStorage.LoadResources(ctx, digest)
+	if err != nil {
+		return nil, err
+	}
+
 	var objects []*unstructured.Unstructured
 
-	ns := &unstructured.Unstructured{}
-	ns.SetName("foo")
-	ns.SetAPIVersion("v1")
-	ns.SetKind("Namespace")
-	ns.SetAnnotations(map[string]string{
-		"created-by": subject.GetNamespace() + "-" + subject.GetName(),
-	})
+	for _, item := range resources.Items {
+		ext := path.Ext(item.Path)
+		ext = strings.ToLower(ext)
+
+		parse := false
+		switch ext {
+		case ".yaml", ".yml":
+			parse = true
+
+		default:
+			klog.Warningf("ignoring non-yaml file %s", item.Path)
+		}
+
+		if !parse {
+			continue
+		}
+		for _, s := range strings.Split(item.Contents, "\n---\n") {
```

Comment:
I wonder if kyaml has a parser that can handle this?

Reply:
It sort of does, but it's not exported: https://github.com/kubernetes-sigs/kustomize/blob/a5b61016bb40c30dd1b0a78290b28b2330a0383e/kyaml/kio/byteio_reader.go#L170

Added a todo